airstack-modules/modules/airstack/social/social.ts

```diff
@@ -111,39 +165,77 @@ export namespace social {
    * @param tokenId erc721 token id
    * @param tokenAddress erc721 token address
    * @param socialUserId social user id (eg: farcasterId)
-  */
-  export function trackSocialProfileOnwershipChangeTransaction(
+   */
+  export function trackSocialProfileOwnershipChangeTransaction(
     block: ethereum.Block,
     transactionHash: string,
     logOrCallIndex: BigInt,
     from: string,
     to: string,
     tokenId: string,
     tokenAddress: string,
-    socialUserId: string,
+    socialUserId: string
   ): void {
-    const chainId = getChainId();
-    const airBlock = getOrCreateAirBlock(chainId, block.number, block.hash.toHexString(), block.timestamp);
-    airBlock.save();
-    const airSocialProfile = getAirSocialProfile(
+    const chainId = getChainId()
+    const airBlock = getOrCreateAirBlock(
       chainId,
-      tokenAddress,
-      tokenId,
-    );
+      block.number,
+      block.hash.toHexString(),
+      block.timestamp
+    )
+    airBlock.save()
+    const airSocialProfile = getAirSocialProfile(chainId, tokenAddress, tokenId)
     if (airSocialProfile == null) {
-      throw new Error("air social profile not found");
+      throw new Error("air social profile not found")
     }
     const airSocialUser = createAirSocialUser(
       chainId,
       airBlock,
       socialUserId,
       to,
-      new Array<string>(),
-    );
-    airSocialProfile.user = airSocialUser.id;
-    airSocialProfile.lastUpdatedAt = airBlock.id;
-    airSocialProfile.save();
-    createAirSocialProfileOnwershipChangeTransaction(
+      new Array<string>()
+    )
+    airSocialProfile.user = airSocialUser.id
+    airSocialProfile.lastUpdatedIndex = updateAirEntityCounter(
```

Comment:
I would recommend to wrap all save call & lastUpdatedIndex in a single function so that its never missed.